src/UI.cpp

```diff
@@ -1729,6 +1728,7 @@ unsigned long getHexValue(const char sTitle[], unsigned long defValue) {
                     LCD.print_P(3, 11, UIStrings::Generic::LESS_SYM);
                 }
                 else {
+                    multiplier = ulpow(16, (digits - cursorPos - 1));
```

Comment:
I believe it was an really long loop when cursorPos == digits, because once you exited entry mode on one of the characters it would just hang.

Consider the case where cursorPos = encValue and encValue =2. It will then enter ulpow with params 16 and -1. The loop within ulpow will increment until i < 2^32 -1 because of an implicit conversion from -1 to unsigned long. So the function will basically end up calculating out 16^(4+billion), which isn't infinite, but certainly will take a long time :)

Does that sound right?